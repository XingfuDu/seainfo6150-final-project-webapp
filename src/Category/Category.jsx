import React from 'react';
import classes from './Category.module.css';
import { Link } from 'react-router-dom';

const Category = () => {
	return (
		<div className={classes.wrapper}>
			<main className={classes.main}>
				<ul className={classes.ul}>
					<li className={classes.li}>
						<Link className={classes.link} to={`/book/1`}>
							Psychology
						</Link>
					</li>
					<li className={classes.li}>
						<Link className={classes.link} to={`/book/2`}>
							Philosophy
						</Link>
					</li>
					<li className={classes.li}>
						<Link className={classes.link} to={`/book/3`}>
							Sociology
						</Link>
					</li>
					<li className={classes.li}>
						<Link className={classes.link} to={`/book/4`}>
							Environment
						</Link>
					</li>
					<li className={classes.li}>
						<Link className={classes.link} to={`book/5`}>
							Science
						</Link>
					</li>
					<li className={classes.li}>
						<Link className={classes.link} to={`/book/6`}>
							Anthropology
						</Link>
					</li>
				</ul>
			</main>
		</div>
	);
};

export default Category;
