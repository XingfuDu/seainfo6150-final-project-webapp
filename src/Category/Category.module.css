.wrapper {
	width: 100%;
	min-height: 100%;
	background-color: #f2e9e4;
	padding: 15px;
    display: flex;
    justify-content: center;
    align-items: center;
}
.main {
	min-height: 100%;
	font-family: Arial, Helvetica, sans-serif;
	font-size: 150%;
}

.ul {
	display: grid;
	grid-template-columns: repeat(3, 1fr);
    list-style-type: none;
	gap: 20px;
}
.li {
    border: 1px solid #ee6c4d;
    border-radius: 15px;
    width: 200px;
    height: 200px;
    text-align: center;
    padding-top: 85px;
    background-color:#ede0d4;
    box-sizing: border-box;
    margin: 30px;
}

.li:hover {
    background-color:#fdf1e6;
}

.link {
    margin: 0 0 20px;
    color: #3d5a80;
    min-width: 50%;
    text-decoration: none;
}