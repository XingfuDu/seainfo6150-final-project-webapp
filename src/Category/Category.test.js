import React from 'react';
import Category from './Category';
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

describe('Category tests', () => {
	it('renders correctly', () => {
		const { container } = render(<BrowserRouter><Category/></BrowserRouter>);
		expect(container).toMatchSnapshot();
	});
});
