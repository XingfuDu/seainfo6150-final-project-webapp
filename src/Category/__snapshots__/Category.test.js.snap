// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category tests renders correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <main
      class="main"
    >
      <ul
        class="ul"
      >
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/1"
          >
            Psychology
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/2"
          >
            Philosophy
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/3"
          >
            Sociology
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/4"
          >
            Environment
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/5"
          >
            Science
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="link"
            href="/book/6"
          >
            Anthropology
          </a>
        </li>
      </ul>
    </main>
  </div>
</div>
`;
